src/main/java/com/amazon/opendistroforelasticsearch/security/DefaultObjectMapper.java

```diff
@@ -214,6 +217,19 @@ public T run() throws Exception {
             return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
                 @Override
                 public T run() throws Exception {
+                    if (jt.getBindings().getTypeParameters().get(0).getRawClass() == InternalUserV6.class) {
+                        JsonNode root = readTree(string);
+                        Iterator<JsonNode> internalUsers = root.elements();
+                        while (internalUsers.hasNext()) {
+                            ObjectNode internalUser = (ObjectNode) internalUsers.next();
+                            JsonNode plainTextPassword = internalUser.get("password");
+                            if (plainTextPassword == null)
+                                continue;
+                            if (plainTextPassword.asText().length() == 0)
```

Comment:
You can directly remove any password if exists? We should not be storing any plaintext password. Only hash should be stored

Reply:
Yes, empty passwords will be removed from here to prevent migration issue caused by empty password for existing domains that have already had internal users with empty passwords.

Reply:
I mean to say, we don't have to explicitly check for empty password. It should not exist at all and even if it exists, it will be empty.

Reply:
I agree. Thanks for clarifying this!